src/main/java/org/apache/sysml/runtime/matrix/data/LibMatrixCuDNN.java

```diff
@@ -506,41 +489,29 @@ public static void maxpooling(GPUContext gCtx, String instName, MatrixObject ima
 		}
 	}
 
-	private static void cudnnMaxpooling(GPUContext gCtx, String instName, Pointer x, cudnnTensorDescriptor xDesc,
+	private static void cudnnMaxpooling(GPUContext gCtx, String instName, Pointer x,
 			Pointer y, int N, int C, int H, int W, int K, int R,
 			int S, int pad_h, int pad_w, int stride_h, int stride_w, int P,
 			int Q) throws DMLRuntimeException {
 		if(LOG.isTraceEnabled()) {
 			LOG.trace("GPU : performMaxpooling" + ", GPUContext=" + gCtx);
 		}
 
-		cudnnPoolingDescriptor poolingDesc = null;
-
-		try {
+		try(LibMatrixCuDNNPoolingDescriptors desc = 
```

Comment:
I like this. You may want to put the `MISC_TIMER_CUDNN_CLEANUP` timer in the `LibMatrixCuDNNPoolingDescriptors` class.